Add encryption option and components for PostgreSQL, MySQL, MongoDB

BlackYoup opened the pull request:
Fixes #145

hsablonniere commented on src/templates/cc-addon-encryption-at-rest-option.js:
```diff
@@ -0,0 +1,25 @@
+import { assetUrl } from '../lib/asset-url.js';
+import { i18n } from '../lib/i18n.js';
+
+const LOGO = assetUrl(import.meta, '../assets/encryption-at-rest.svg');
```
The current convention is `camelCaseNameOfTheImageSvg` with a `Svg` suffix.

BlackYoup replied:
Should be fixed now

hsablonniere commented on src/addon/cc-addon-postgresql-options.js:
```diff
@@ -0,0 +1,83 @@
+  _getFormOptions () {
+    return this.options.map((option) => {
+      switch (option) {
+        case 'encryption': return ccAddonEncryptionAtRestOption(ENCRYPTION_DOCUMENTATION_URL, this.encryptionPrice);
+        default: return null;
```
Not a great fan of one liners like this. Is ESLint OK with that? If so, let's keep it and I'll update the ESLint config :p

BlackYoup replied:
Yes, eslint is ok :p

hsablonniere replied:
OK so let's keep it like that, we'll update ESLint later.

hsablonniere commented on src/addon/cc-addon-postgresql-options.js:
```diff
@@ -0,0 +1,83 @@
+  }
+
+  _getFormOptions () {
+    return this.options.map((option) => {
+      switch (option) {
+        case 'encryption': return ccAddonEncryptionAtRestOption(ENCRYPTION_DOCUMENTATION_URL, this.encryptionPrice);
+        default: return null;
+      };
+    }).filter((option) => option !== null);
```
In those situations, you can align the `.map` and `.filter`

BlackYoup replied:
Updated, please double check to see if I correctly understood you

hsablonniere replied:
This is what I expected.

hsablonniere commented on src/addon/cc-addon-mysql-options.js:
```diff
@@ -0,0 +1,83 @@
+
+  static get properties () {
+    return {
+      options: { type: Array, attribute: 'options' },
```
No need for `attribute` there.

hsablonniere commented on src/addon/cc-addon-mongodb-options.js:
```diff
@@ -0,0 +1,83 @@
+
+  static get properties () {
+    return {
+      options: { type: Array, attribute: 'options' },
```
No need for `attribute` here.

hsablonniere commented on src/addon/cc-addon-postgresql-options.js:
```diff
@@ -0,0 +1,83 @@
+
+  static get properties () {
+    return {
+      options: { type: Array, attribute: 'options' },
```
No need for `attribute` here.

hsablonniere reviewed (changes requested):
I haven't looked at the result in a browser yet.

hsablonniere commented:
I didn't understood at first that options were opt-in with `options`.
We need a story to see what the component looks like when `options` is an empty array.
Will we need an empty message?

I see `encryptionEnabled` in:

* the JSDoc header
* the `static get properties () {}`
* the `constructor`

but it's never used in the component template and never used by the story.

BlackYoup commented:
> I didn't understood at first that options were opt-in with options.

Yes, options might be different across add-ons plans

> We need a story to see what the component looks like when options is an empty array.

Yeah, I didn't make one because I assumed that you need to provide at least one option, otherwise this option selection shouldn't even be displayed. I can indeed add a `there are no options` text if you think it's worth adding such a case.

I fixed the `encryptionEnabled` boolean not correctly displayed and added a story with the , sorry for the silly mistake.

hsablonniere commented:
> This option is currently free but will then be billed 10.0% of the plan's price. If this option is enabled, it will be billed €10.00 per month.

I still think it's a bit confusing.

hsablonniere commented:
About the translation, @urcadox suggests:

> This option is currently free. In the future, it will be billed 10% of the plan price, which amounts to €10 per month here.

hsablonniere commented:
I think we should move `options` from `String[]` to `Option[]` where `Option` has a `name`, a `enabled` boolean and some custom options like `price` for encryption or `flavor` for apm and kibana.
